src/ConfigEditor/ConfigEditor.test.tsx

```diff
@@ -51,287 +52,309 @@ jest.mock('@grafana/runtime', () => {
+            database: 'abcd',
           },
-        }}
-      />
-    );
-
-    // the dbUser update is delegated to the onChange function
-    await waitFor(() =>
-      expect(onChange).toHaveBeenCalledWith({
-        ...props.options,
-        url: 'foo.a.b.c:123/',
-        jsonData: {
-          ...props.options.jsonData,
-          dbUser,
-          useServerless: true,
-          useManagedSecret: true,
-          managedSecret: serverlessSecret,
-          workgroupName,
-        },
-      })
-    );
-  });
+        });
+      });
+
+      it('should show the cluster identifier and the db user', async () => {
+        const onChange = jest.fn();
+        render(
+          <ConfigEditor
+            {...props}
+            onOptionsChange={onChange}
+            // setting the managedSecret will trigger the secret retrieval
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                useServerless: false,
+                useManagedSecret: true,
+                managedSecret: provisionedSecret,
+              },
+            }}
+          />
+        );
+
+        // the clusterIdentifier and dbUser update is delegated to the onChange function
+        await waitFor(() =>
+          expect(onChange).toHaveBeenCalledWith({
+            ...props.options,
+            url: 'bar.d.e.f:456/',
+            jsonData: {
+              ...props.options.jsonData,
+              dbUser,
+              useServerless: false,
+              useManagedSecret: true,
+              managedSecret: provisionedSecret,
+              clusterIdentifier,
+            },
+          })
+        );
+      });
+
+      it('should show the dbUser', async () => {
+        const onChange = jest.fn();
+        render(
+          <ConfigEditor
+            {...props}
+            onOptionsChange={onChange}
+            // setting the managedSecret will trigger the secret retrieval
+            options={{
+              ...props.options,
+              jsonData: {
+                ...props.options.jsonData,
+                useServerless: true,
+                useManagedSecret: true,
+                managedSecret: serverlessSecret,
+                workgroupName,
+              },
+            }}
+          />
+        );
+
+        // the dbUser update is delegated to the onChange function
+        await waitFor(() =>
+          expect(onChange).toHaveBeenCalledWith({
+            ...props.options,
+            url: 'foo.a.b.c:123/',
+            jsonData: {
+              ...props.options.jsonData,
+              dbUser,
+              useServerless: true,
+              useManagedSecret: true,
+              managedSecret: serverlessSecret,
+              workgroupName,
+            },
+          })
+        );
+      });
+    });
+  }
+  run('ConfigEditor with new form styling disabled');
+  config.featureToggles.awsDatasourcesNewFormStyling = true;
```

Comment:
in line 57 this toggle is already set to true. also not sure whether this will have an effect since `@grafana/runtime` is mocked above?

Reply:
Ok so there indeed is a problem with these tests, and TIL about Jest's setup and teardown lifecycle 👩🏻‍🎓
Since jest "[executes all describe handlers in a test file before it executes any of the actual tests](https://jestjs.io/docs/setup-teardown)", the value of the feature toggle will always be true (or whatever value we put last), since we overwrite it globally. 
The only way I found to circumvent this is to separate "toggle disabled" and "toggle enabled" describe blocks and add 
```
beforeAll(() => {
        valueOfTheToggle = whatever
})
``` 

inside each block. Maybe there's a better way that I can't see right now (or just copying the tests lol)🤔